R/moduleInstall.R

```diff
@@ -1,12 +1,338 @@
+  assertValidJASPmodule(modulePkg)
+
+  r <- getOption("repos")
+  r["CRAN"] <- repos
+  options(repos = r)
+
+  if (!(force || md5SumsChanged(modulePkg, moduleLibrary))) {
+    moduleName <- getModuleInfo(modulePkg)[["Package"]]
+    if (dir.exists(file.path(moduleLibrary, moduleName))) {
+      print(sprintf("Nothing changed according to md5sums, not reinstalling %s.", moduleName))
+      return("succes!")
+    } else {
+      print(sprintf("Checksums exist for %s but the package is missing, installing anyway!", moduleName))
+    }
+  }
+
+  return(
+    pkgbuild::with_build_tools( 
+    {
+      if (hasRenvLockFile(modulePkg)) installJaspModuleFromRenv(       modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg)
+      else                            installJaspModuleFromDescription(modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg)
+    }, 
+    required=FALSE )
+  )
+}
+
+
+
+installJaspModuleFromRenv <- function(modulePkg, libPathsToUse, moduleLibrary, repos, onlyModPkg, prompt = interactive()) {
+
+  print(sprintf("Installing module with renv. installJaspModuleFromRenv('%s', c(%s), '%s', '%s', %s)",
+                modulePkg, paste0("'", libPathsToUse, "'", collapse = ", "), moduleLibrary, repos, onlyModPkg))
+
+  renv::consent() #Consent to doing stuff in certain paths: https://rstudio.github.io/renv/reference/consent.html We've changed at least the root and the cache so hopefully that means it won't be changing stuff in the locations it mentions
+  options(renv.verbose=TRUE) # More feedback wanted although this seems to do little
+  options(renv.config.install.verbose=TRUE)
+
+  if (!dir.exists(moduleLibrary))
+    if (!dir.create(moduleLibrary))
+      stop("failed to create moduleLibrary!")
+
+  # ensure this starts with a ., otherwise it's picked up by renv "helpers"
+  moduleLibraryTemp <- file.path(moduleLibrary, ".renv_temp")
+
+  if (!dir.exists(file.path(moduleLibraryTemp, "renv")))
+    dir.create(   file.path(moduleLibraryTemp, "renv"), recursive = TRUE)
+
+  lockFileModule <- file.path(modulePkg,         "renv.lock")
+  lockFileTemp   <- file.path(moduleLibraryTemp, "renv.lock")
+  file.copy(from = lockFileModule, to = lockFileTemp, overwrite = TRUE)
+
+  setupRenv(moduleLibrary)
+
+  # TODO: unclear whether this is necessary within JASP, or just within Rstudio.
+  # renv must be unaware of any other libPaths than the cache and the directory designated for the module.
+  # if there are other libPaths then renv may reuse pkgs from those other libPaths
+  # it does copy those pkg to the cache before symlinking them
+  # inspired by https://stackoverflow.com/a/36873741/4917834
+  # it does appear to be necessary within rstudio and when the pkgs from jasp-required-files are present in a libPath
+  old.lib.loc <- .libPaths()
+  on.exit(assign(".lib.loc", old.lib.loc,   envir = environment(.libPaths)))
+          assign(".lib.loc", moduleLibrary, envir = environment(.libPaths))
```

Comment:
We should probably test to what extent it is necessary to keep this in there. I added it originally so renv wouldn't copy packages from my local R library, in other words, force renv to use whatever we give it. But if that's already guaranteed by JASP through other ways then maybe we should delete this.

Reply:
yeah I think there is a whole bunch more stuff that can be scrapped. Also the whole required-modules stuff can be gone. Because renv fixes all of that now.

But I first wanted to get a functional version up. After that I will start cleaning up the dynamic module code and then this can possibly be taken with. Or separately for that matter.